include/pybind11/pytypes.h

```diff
@@ -491,7 +491,7 @@ class tuple : public object {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
     size_t size() const { return (size_t) PyTuple_Size(m_ptr); }
-    detail::tuple_accessor operator[](size_t index) const { return detail::tuple_accessor(ptr(), index); }
```

Comment:
It has no effect to the args issue, but I think it's a missing spot when tuple_accessor constructor changed from taking a PyObject * to handle.  Giving it this->ptr() is the same as *this.  But list::operatorp[]() changed to send *this, so for consistency I think here we should use *this too.